### PR TITLE
fix: Invalidation for file cache in Range.ml

### PR DESCRIPTION
**Pull request by nmote**
This addresses a DeepSemgrep crash when run on the `NodeBB` repo (see test plan), but it's probably possible to trigger this crash in other ways including without DeepSemgrep.

Basically, the issue is that collisions in tmp file names can lead to outdated information available in the cache. If we try to read, a substring starting at, say character 10, but the temp file that is cached is only 5 characters long, we get a crash.

Test plan: See https://github.com/returntocorp/pfff/pull/575

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by nmote** [comment]
Don't merge -- I'll update once the pfff PR is merged

**Review by aryx — APPROVED** [review]

**Pull request merged**
